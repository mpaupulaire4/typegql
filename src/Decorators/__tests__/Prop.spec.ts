import { Prop, PropsMetadataKey } from '../'

describe('Prop Decorator', () => {

  it('should return a function', () => {
    expect(typeof Prop()).toBe('function')
  })

  it('Metatdata should be a map of prop name to name of type constructor', () => {
    class Test {
      @Prop()
      string: string;
      @Prop()
      number: number;
      @Prop()
      bool: boolean;

      @Prop()
      static stringStatic: string;
      @Prop()
      static numberStatic: number;
      @Prop()
      static boolStatic: boolean;
    }
    expect(Reflect.getMetadata(PropsMetadataKey, Test)).toMatchSnapshot()
  })

  it('Metadata should map prop to "MyType', () => {
    class Test {
      @Prop('MyType!')
      prop: string;
    }
    expect(Reflect.getMetadata(PropsMetadataKey, Test)).toMatchSnapshot()
  })

})