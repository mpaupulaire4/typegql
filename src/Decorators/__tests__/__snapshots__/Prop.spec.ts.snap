// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prop Decorator Metadata should map prop to "MyType 1`] = `
Map {
  "prop" => "MyType!",
}
`;

exports[`Prop Decorator Metatdata should be a map of prop name to name of type constructor 1`] = `
Map {
  "string" => "String",
  "number" => "Number",
  "bool" => "Boolean",
  "stringStatic" => "String",
  "numberStatic" => "Number",
  "boolStatic" => "Boolean",
}
`;
